lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -1162,6 +1258,17 @@ FunctionEntrypointAnalyzer::createFakeFunction(llvm::BasicBlock *Entry) {
   return FakeFunction.extractFunction();
 }
 
+void FunctionEntrypointAnalyzer::initializeFakeFunctions() {
+  for (const auto &Function : Binary->Functions) {
+    if (Function.Type != FunctionTypeValue::Fake)
+      continue;
+
+    auto &Summary = Oracle.at(Function.Entry);
+    revng_assert(Summary.Type == FunctionTypeValue::Fake);
+    Summary.FakeFunction = createFakeFunction(GCBI->getBlockAt(Function.Entry));
+  }
+}
+
```

Comment:
You have a `model::Function`, you get its type and its address and create an entry into the EFA cache.
This really sounds like importing to me.
Please merge them.

Reply:
Sorry, I recalled only now – while changing this – the actual reason behind this. `createFakeFunction` is part of `Analyzer` method. Originally, I had tried to isolate this and make `createFakeFunction` a separate static function, but no success on it (it requires too much stuff that is part of `FunctionEntrypointAnalyzer` object), that's why I eventually split the two things.

Reply:
I don't understand.
Can't you move the body of `initializeFakeFunctions` at the end of `importModel`?

Reply:
I cannot. `importModel` is a public method of the local EFA cache (`FunctionAnalysisResults`), whereas `initializeFakeFunctions` needs to be a method of `FunctionEntrypointAnalyzer`, since it needs to call `FunctionEntrypointAnalyzer::createFakeFunction` to create the fake function.

Reply:
This design is a bit smelly.
Anyway, do not read the model then, read the list of functions tracked by the oracle and create a function for each fake function in there.
Otherwise we're importing the model in two steps, which is confusing.

Reply:
Uhmmm, no, I need to read the model. I don't have a list of functions tracked by the oracle. Note that this is actually done also in `FunctionEntrypointAnalyzer::recoverCFG`, `FunctionEntrypointAnalyzer::interproceduralPropagation` and `FunctionEntrypointAnalyzer::finalizeModel`.

Reply:
This has been fixed by making `importModel` a method of `FunctionEntrypointAnalyzer` (and not of the cache). As a result:
1) Now the local EFA cache is constructed as "dumb", like empty (as it is being done in the current version of EFA in develop).
2) `Analyzer.importModel` is a public method called in order to import prototype of existing functions from model, re-create fake functions (`initializeFakeFunctions` has been dropped, no need for a new function) and to import dynamic functions as well.

Reply:
Sounds good!